src/main/java/school/faang/user_service/validation/goal/GoalInvitationValidator.java

```diff
@@ -0,0 +1,69 @@
+package school.faang.user_service.validation.goal;
+
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import school.faang.user_service.dto.goal.GoalInvitationDto;
+import school.faang.user_service.entity.goal.GoalInvitation;
+import school.faang.user_service.exception.DataValidationException;
+import school.faang.user_service.repository.UserRepository;
+import school.faang.user_service.repository.goal.GoalInvitationRepository;
+import school.faang.user_service.repository.goal.GoalRepository;
+
+@Component
+@RequiredArgsConstructor
+public class GoalInvitationValidator {
+    private static final long MAX_ACTIVE_GOALS = 3;
+
+    private final GoalRepository goalRepository;
+    private final UserRepository userRepository;
+    private final GoalInvitationRepository goalInvitationRepository;
+
+    public void validateCreateInvitation(GoalInvitationDto invitationDto) {
+        Long inviterId = invitationDto.getInviterId();
+        Long invitedUserId = invitationDto.getInvitedUserId();
+
+        // Условие: в приглашении есть приглашаемы и приглашающие пользователи
+        if (inviterId == null || invitedUserId == null) {
+            throw new DataValidationException("One of the participants was not indicated in the invitation");
+        }
+        // Условие: они не один и тот же человек
+        if (inviterId.equals(invitedUserId)) {
+            throw new DataValidationException("Cannot specify the same ID for the [inviter] and [invitedUser] fields");
```

Comment:
throw new DataValidationException("Cannot specify the same ID:"+ invitedUserId +" for the [inviter] and [invitedUser] fields");